### PR TITLE
Fix validation #2546,#2555, searchbar bug #2640, avatar UI #2638

### LINKED ISSUE #2546: Login and Password reset: Improve user security
The system is not hardened against brute-force attacks. There a few small things we can do. Please open relevant issues in the server as well. 

* Require user passwords to have at least 8 characters and at least one special character, one number and text characters
* Add Recaptcha for second log-in attempts
* Add Recaptcha for password resets (forgot password)

Question: How can this be made working with the server? If someone could just call the API from another domain we don't gain anything.

Also, this needs to work with the Android app. Please collaborate with the Android team.

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #2546, #2638, #2555, #2640

Changes: 
- [x] Avatar on click interaction
- [x]  Require user passwords to have at least 8 characters and at least one special character, one number and text characters
 - [x] Add Recaptcha for second log-in attempts
 - [x] Add Recaptcha for password resets (forgot password)
- [x] Username less than or equal to 51 characters.
- [x] Fix email autofill in the search bar

Demo Link: https://pr-2639-fossasia-susi-web-chat.surge.sh

Screenshots of the change:

Screenshot 2019-07-20 at 6 20 01 PM

### CODE REVIEW

**Comment by fragm3** [comment]
Note: I have intentionally left the comment in password change as for now users are following old regex and if we change it, they won't be able to change password.

File: src/components/Auth/ChangePassword/ChangePassword.react.js [hunk 1]
```diff
@@ -104,18 +124,15 @@ class ChangePassword extends Component {
     switch (event.target.name) {
       case 'password': {
         const password = event.target.value.trim();
-        const passwordError = !(
-          password &&
-          password.length >= 6 &&
-          password.length <= 64
-        );
+        // const passwordError = !passwordRegEx.test(password)
```
Comment (akshatnitd) [inline src/components/Auth/ChangePassword/ChangePassword.react.js]: Do we really require regex in current password? I think we can remove it.

File: src/components/Settings/AccountTab.react.js [hunk 2]
```diff
@@ -22,6 +22,7 @@ import styled from 'styled-components';
 import { setUserSettings, uploadAvatar } from '../../apis';
 import defaultAvatar from '../../images/defaultAvatar.png';
 import CircularProgress from '@material-ui/core/CircularProgress';
+import { userNameRegEx } from '../../utils/regex';
```
Comment (akshatnitd) [inline src/components/Settings/AccountTab.react.js]: Can you use this constant wherever applicable to remove redundacy? Like for eg - Signup modal.
Same for passwordRegex.

**Comment by fragm3** [comment]
@akshatnitd made the changes

**Pull request merged**